atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -351,6 +359,16 @@ private void addSnapshotListener() {
                 .build()));
   }
 
+  private void onSnapshotReceiveCompletedOrAborted() {
+    if (pendingSnapshot != null) {
+      pendingSnapshot.abort();
+    }
+    // Listeners should be notified whether snapshot is committed or aborted. Otherwise they can
+    // wait for ever.
+    raft.notifySnapshotReplicationCompleted();
```

Comment:
:question: Is it really safe to make no distinction between successful completion and abort?

Reply:
For our current use case, yes. If we need to distinguish in future we can udpate the interface then.